From 62297e906149f9d2ca7b187a60f7c5ab99ef5dd5 Mon Sep 17 00:00:00 2001
From: Thomas Petazzoni <thomas.petazzoni@bootlin.com>
Date: Thu, 26 Dec 2024 21:39:05 +0100
Subject: [PATCH] cldap_ping.c: add missing <sys/types.h> include
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fixes:

cldap_ping.c: In function ‘read_dns_string’:
cldap_ping.c:72:37: error: ‘u_char’ undeclared (first use in this function)

when building with the musl C library, but even with glibc u_char is
defined in <sys/types.h>, it happens to work with glibc <sys/types.h>
gets included by another header.

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@bootlin.com>
Upstream: https://lore.kernel.org/linux-cifs/20241226204235.2311371-1-thomas.petazzoni@bootlin.com/
---
 cldap_ping.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/cldap_ping.c b/cldap_ping.c
index 9183b27..a603be3 100644
--- a/cldap_ping.c
+++ b/cldap_ping.c
@@ -20,6 +20,7 @@
 #include <talloc.h>
 #include <string.h>
 #include <sys/socket.h>
+#include <sys/types.h>
 #include <arpa/inet.h>
 #include <unistd.h>
 #include <resolv.h>
-- 
2.47.0

